src/integration/java/io/pinecone/helpers/AssertRetry.java

```diff
@@ -21,14 +24,14 @@ public static void assertWithRetry(AssertionRunnable assertionRunnable, int back
                 success = true;
```

Comment:
Can we update this function to the following:
```java
    public static void assertWithRetry(AssertionRunnable assertionRunnable, int backOff) throws InterruptedException {
        int retryCount = 0;
        boolean success = false;
        String errorMessage = null;

        while (retryCount < maxRetry && !success) {
            try {
                assertionRunnable.run();
                success = true;
            } catch (AssertionError | ExecutionException | IOException e) {
                retryCount++;
                delay*=backOff;
                Thread.sleep(delay);
                errorMessage = e.getLocalizedMessage();
            }
        }

        if (!success) {
            throw new AssertionError(errorMessage);
        }
    }
```

I dont have strong opinions about adding backOff before or after and adding the PineconeException to the signature but this will make sure the assert throws an error after all of the retries.